sdk/client_errors.go

```diff
@@ -0,0 +1,80 @@
+package sdk
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"text/template"
+)
+
+const concurrencyIssueMessage = "operation failed due to concurrency issue but can be retried"
+
+// APIError represents an HTTP error response from the API.
+type APIError struct {
+	Message    string `json:"message"`
+	StatusCode int    `json:"statusCode"`
+	Method     string `json:"method"`
+	URL        string `json:"url"`
+	TraceID    string `json:"traceId,omitempty"`
+}
+
+// IsConcurrencyIssue returns true if the underlying API error is a concurrency issue.
+// If so, the operation can be retried.
+func (r APIError) IsConcurrencyIssue() bool {
```

Comment:
I imagine there might be other kinds of errors that are not entirely related to concurrent issues.
As a consumer, I'm more interested in whether a request that ended up with an error can be retried rather than whether it was caused by concurrent requests (which honestly should be 409).

Maybe we could introduce a property similar to the one you can find in aws sdk (`IsErrorRetryable` - https://github.com/aws/aws-sdk-go-v2/blob/main/aws/retry/retryable_error.go#L15)

Reply:
that's a great thought! I agree :)